GitUI/UserControls/RevisionGrid/FilterInfo.cs

```diff
@@ -181,6 +181,16 @@ public bool HasFilter
                 || !string.IsNullOrWhiteSpace(BranchFilter);
         }
 
+        /// <summary>
+        /// Has any filters which might hide branches or tags.
+        /// </summary>
+        public bool HasRefsFilter
+        {
+            get => HasFilter
+                || IsShowCurrentBranchOnlyChecked
+                || (ByCommitsLimit && HasCommitsLimit);
```

Comment:
Note: The default 100K limit will hide commits in for instance Linux repo, so technically all 'filters' could hide commits. The settings value could be changed too.
I believe this is good enough.

Reply:
I have tested this: If the number of commits is limited, the Left Panel contains branches and tags which are not visible in the RevisionGrid.
Do you mean `|| AppSettings.MaxRevisionGraphCommits > 0` should be added?

BTW: Bad UX:
![Filter_Limit_is_reset](https://user-images.githubusercontent.com/36601201/209740169-1e599b4e-ec51-4d4c-9963-d0a20bf05458.gif)

Reply:
I say that we ignore the case where default limit do not load all refs. 
What is bad with the ui?

Reply:
> I say that we ignore the case where default limit do not load all refs.

It is clearer and simpler with `HasCommitsLimit`, which already takes the default limit into account.

---

> What is bad with the ui?

Con, main point: The user input is lost when the Checkbox is unticked.
Pro: The effective value is displayed.

Reply:
> > I say that we ignore the case where default limit do not load all refs.
> 
> It is clearer and simpler with `HasCommitsLimit`, which already takes the default limit into account.

But then the check is meaningless, as isFiltering will be true except when users have set commits limit to 0, which I would expect is very uncommon.
Then it is better to remove isFiltering for side panel completely.
As this is for 4.0, I suggest to just set the value in RefreshLeftPanel() with a comment and cleanup in master. Maybe rendering is done twice there now (a default rendering applying style must be done, maybe there is a check for visibility too.)

> > What is bad with the ui?
> 
> Con, main point: The user input is lost when the Checkbox is unticked. Pro: The effective value is displayed.

(Not really UI as I would define it but possibly confusing).
This was a decision I took when working with this. Can be discussed.
Welcome to the club of maintainers burnt out by filters, first RussKie, then me, now you too?

Reply:
> But then the check is meaningless.

Perhaps for most of the users. But the check is not very expensive and the result is correct.
Since I had to wonder why `Blame this version` was disabled, I have turned off the default limit.
Filtering always is most safe of course and would reduce the code size a little.
Shall I remove `isFiltering`? There are only a few locations.
(But I will keep the tests added for `HasFilter` and the more consistent test setup.)

Reply:
> Shall I remove isFiltering? There are only a few locations.

I would like to see that, the side panel code is complicated as it is.

Reply:
I have removed `isFiltering` and `FilterInfo.HasFilter`